src/main/java/seedu/address/logic/commands/MailAllCommand.java

```diff
@@ -17,24 +19,54 @@
 
     public static final String COMMAND_WORD = "mail-all";
 
+    public static final String DESKTOP_NOT_SUPPORTED_MESSAGE = "The desktop you are using is not supported";
+
+    public static final String NO_EMAILS_SPECIFIED = "No email addresses specified.";
+
+    public static final String URI_SYNTAX_ERROR_MESSAGE = "The URI syntax used is incorrect";
+
     private static final String MESSAGE_SUCCESS = "Opening app for bulk emailing";
 
     /**
-     * Executes the mailall command.
+     * Executes the mail-all command.
      * @param model {@code Model} which the command should operate on.
      * @return A {@code CommandResult} object specifying the success message.
      * @throws CommandException if the command execution goes unexpected.
      */
     @Override
     public CommandResult execute(Model model) throws CommandException {
-        List<Person> lastShownList = model.getFilteredPersonList();
-        List<Email> emails = lastShownList.stream().map(Person::getEmail).distinct().collect(Collectors.toList());
+        String[] emailStrings = retrieveEmailStrings(model);
```

Comment:
Just one question, what's the rationale behind using a String list/array instead of an Email list/array?